rdwatch/core/views/model_run.py

```diff
@@ -637,3 +650,35 @@ def get_downloaded_annotations(request: HttpRequest, id: UUID4, task_id: str):
             'Content-Disposition'
         ] = f'attachment; filename="{annotation_export.name}.zip"'
         return response
+
+
+@router.post('/start_upload_processing')
+def start_model_run_upload_processing(
+    request: HttpRequest, upload_data: ModelRunUploadSchema
+):
+    zipfile_upload = signing.loads(upload_data.zipfileKey)
+
+    with transaction.atomic():
+        upload = ModelRunUpload.objects.create(
+            title=upload_data.title.strip(),
+            region=upload_data.region.strip(),
+            performer=upload_data.performer.strip(),
+            zipfile=zipfile_upload['object_key'],
+            private=upload_data.private,
+        )
+        if not upload.title:
+            raise ValidationError('Invalid model run title')
+        if not default_storage.exists(upload.zipfile.name):
+            raise ValidationError('Invalid file name provided')
+
+        task = process_model_run_upload_task.delay(upload.id)
```

Comment:
Theoretically (although unlikely in practice), the transaction creating the `ModelRunUpload` could still be running when the celery task starts -

```suggestion
        task = process_model_run_upload_task.delay_on_commit(upload.id)
```

Reply:
So `delay_on_commit` doesn't return a task ID. I'm thinking of updati this code to instead save the upload task, call `delay_on_commit()`, and have the task itself write its ID to the upload instance. Instead of returning the task ID, we will return the upload instance ID, and have that query the task status.